tests/thread_race/Makefile

```diff
@@ -0,0 +1,5 @@
+include ../Makefile.tests_common
+
+DISABLE_MODULE += auto_init
+
```

Comment:
I forgot this part. I think this test should be run by the CI on hardware. Can you please add:

```makefile
TEST_ON_CI_WHITELIST += all
```

just here?

Reply:
Added support for CI hardware testing.